tests/RobotFramework/tests/cumulocity/software_management/software.robot

```diff
@@ -69,6 +69,44 @@ Manual software_update operation request
     ...    expected_status=failed
     ...    c8y_fragment=c8y_SoftwareUpdate
 
+Operation log uploaded automatically with auto_log_upload setting as on-failure
+    Execute Command    tedge config set c8y.operations.auto_log_upload on-failure
+    Restart Service    tedge-mapper-c8y
+
+    # Validate that the operation log is NOT uploaded for a successful operation
+    ${OPERATION}=    Install Software        c8y-remote-access-plugin
+    Operation Should Be SUCCESSFUL           ${OPERATION}    timeout=60
```

Comment:
I would parametrize the timeouts as well, If timeouts are common across multiple test cases, it make sence to parameterize them to ensure consistency and easy modification, for example:
```
*** Variables ***
${TIMEOUT}    60

*** Test Cases ***
Operation log uploaded automatically with default auto_log_upload setting as onFailure
    Operation Should Be SUCCESSFUL           ${OPERATION}    timeout=${TIMEOUT}
    ...

Operation log uploaded automatically with default auto_log_upload setting as always
    Operation Should Be SUCCESSFUL           ${OPERATION}    timeout=${TIMEOUT}
```